feat: UTC-70: DeletedRow functionality

mcanu opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Adds DeletedRow model, to enable serializing model rows we need to delete but want to keep stored just in case.

mcanu committed:
Added DeletedRow functionality'

jombooth commented on label_studio/core/models.py:
```diff
@@ -42,3 +44,39 @@ class AsyncMigrationStatus(models.Model):
+    Useful for using as backup for deleted rows, in case we need to restore them.
+    """
+
+    model = models.CharField(max_length=255)   # tasks.task, projects.project, etc.
```
one little suggestion here - we have some fairly long model paths already, and this field won't be populated based on user input. I think we can safely use TextField instead to get rid of the max_length, or just set it to something much larger like 1024.

jombooth reviewed (approved):
awesome, I particularly like the `bulk_serialize_and_create` class method you provided 🔥

mcanu committed:
Increased char limit for model

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15451630590)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15451630590

mcanu committed:
Merge branch 'develop' into 'fb-utc-70'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15451630590

hakan458 commented on label_studio/core/models.py:
```diff
@@ -42,3 +44,39 @@ class AsyncMigrationStatus(models.Model):
+
+    @classmethod
+    def serialize_and_create(cls, model, **kwargs) -> 'DeletedRow':
+        data = json.loads(serializers.serialize('json', [model]))[0]
```
this works for any model, and retains all fields, etc? Is there anything that would not be supported?

hakan458 replied:
Also have you tried deserializing?

jombooth replied:
To this point, it would be great to test `Comment` specifically since that's our most immediate purpose for this. No need to write a unit test - just checking it locally is fine.

mcanu replied:
I've added deserialization to tests

mcanu replied:
@jombooth tested it with Comment and it works 👍

mcanu committed:
Added deserialize to test
